docs/general.md

```diff
@@ -0,0 +1,133 @@
+# General Debugr architecture
+
+The Debugr core is very simple and only does a couple of things.
+Most of the actual work is delegated to various plugins. This document
+describes the core concepts of how the Debugr core works.
+
+(todo: explain log entries, plugin types, log levels)
+
+## Log Entries
+
+Log entries are objects which represent data to be logged, along with some
+additional metadata. Log entries can be _generic_, meaning the data has no
+predefined semantics, or they can be _specialised_ to represent a specific
+type of event or information. Whenever you call `logger.log()` or any of the
+shortcuts a new log entry is created. Some plugins will produce entries
+from e.g. a library they integrate with, and other plugins will consume
+entries in order to produce some kind of output.
+
+The shape of a log entry looks like this:
+
+```typescript
+export type LogEntry = {
+  type?: EntryType;
+  level: LogLevel;
+  globalContext: TGlobalContext;
+  taskContext?: TTaskContext;
+  message?: string;
+  data?: Record<string, any>;
+  error?: Error;
+  ts: Date;
```

Comment:
readonly date

Reply:
ja vim, v tech typech v dokumentaci je spousta veci zjednodusenejch, kdyz treba tady napisu `ImmutableDate`, budu vzapeti nekde muset vysvetlit co to je, a tim ta dokumentace hrozne nabobtna.. ale jestli myslis ze je to potreba..

Reply:
fair

Reply:
resp. fair, ale neodpovida to pak tolik realite